core/src/main/scala/org/apache/spark/internal/config/Status.scala

```diff
@@ -70,4 +70,11 @@ private[spark] object Status {
       .version("3.0.0")
       .booleanConf
       .createWithDefault(false)
+
+  val DISK_STORE_DIR_FOR_STATUS =
+    ConfigBuilder("spark.appStatusStore.diskStore.dir")
+      .doc("Local directory where to store app status that couldn't fit in memory")
+      .version("3.3.0")
```

Comment:
I guess this should be `3.4.0` because Today is the feature freeze date for Apache Spark 3.3.0 and this PR arrives a little late for review.

Reply:
@dongjoon-hyun thanks for the review!
I'm wondering if it's possible to include this in `3.3.0`. Here are my two cents:
1. the community doesn't pay attention to Spark's troubleshooting ability for a while. If we can deliver this feature earlier, it could give a signal that the community starts to improve the debuggability and it can attract others to contribute (earlier).
2. as you may see, this PR aims to reduce the impact on the driver and introduces useful features (i.e. show AQE plan changes). Such as separate listener, separate event queue, disk store instead of memory, rest API instead of UI. I know you have concerns about the disk space. I think it's something we can resolve.